components/org.wso2telco.analytics.sparkUdf/src/main/java/org/wso2telco/analytics/sparkUdf/service/BillUpdaterService.java

```diff
@@ -1,89 +1,93 @@
 package org.wso2telco.analytics.sparkUdf.service;
 
+import java.math.BigDecimal;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
 import org.joda.time.LocalDate;
 import org.killbill.billing.client.KillBillClient;
 import org.killbill.billing.client.KillBillClientException;
 import org.killbill.billing.client.KillBillHttpClient;
 import org.killbill.billing.client.RequestOptions;
+import org.killbill.billing.client.model.Account;
 import org.killbill.billing.client.model.Invoice;
 import org.killbill.billing.client.model.InvoiceItem;
 import org.wso2telco.analytics.sparkUdf.configProviders.ConfigurationDataProvider;
 
-import java.math.BigDecimal;
-import java.util.Date;
-import java.util.List;
-import java.util.UUID;
-
 /**
  * @author dilan
  */
 public class BillUpdaterService {
 
 
-    public void updateBillForMonth(String accountId, Date date, String description, int amount) {
-        KillBillHttpClient killBillHttpClient = null;
-        KillBillClient killBillClient = null;
-
-        try {
-            ConfigurationDataProvider dataProvider = ConfigurationDataProvider.getInstance();
-
-            killBillHttpClient = new KillBillHttpClient(dataProvider.getUrl(), dataProvider.getUname(),
-                    dataProvider.getPassword(), dataProvider.getApiKey(), dataProvider.getApiSecret());
-
-            killBillClient = new KillBillClient(killBillHttpClient);
-
-            List<Invoice> invoices = killBillClient.getInvoicesForAccount(UUID.fromString(accountId), RequestOptions.empty());
-            Invoice invoice = getInvoiceForMonthFromList(invoices, date);
-            if (invoice != null) {
-                updateInvoice(invoice, description, amount, killBillClient);
-            }
-
-        } catch (Exception e) {
-
-        } finally {
-            if (killBillClient != null) {
-                killBillClient.close();
-            }
-            if (killBillHttpClient != null) {
-                killBillHttpClient.close();
-            }
-        }
-    }
-    /*
-     *
+	private static ConfigurationDataProvider dataProvider=null;
+	private static KillBillHttpClient killBillHttpClient;
+	private static KillBillClient killBillClient;
+	private static UUID invoiceItemId=null;
+
+	@SuppressWarnings("deprecation")
+	public String l(String accountId,int year,int month,String description,int amount){
+		Account account=null;
+		try {
+			dataProvider=ConfigurationDataProvider.getInstance();
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.wso2telco.com:9000/coding_rules#rule_key=squid%3AS2696)